packages/terra-folder-tree/src/subcomponents/FolderTreeItem.jsx

```diff
@@ -112,6 +123,48 @@ const FolderTreeItem = ({
     }
   };
 
+  const handleKeyDown = event => {
+    switch (event.nativeEvent.keyCode) {
+      case KeyCode.KEY_RETURN:
+      case KeyCode.KEY_SPACE:
+        event.preventDefault();
+
+        if (isFolder) {
+          handleToggle(event);
+        } else {
+          onClick(event);
+        }
+        break;
+      case KeyCode.KEY_LEFT: {
+        event.preventDefault();
+        if (event.metaKey) { break; }
+
+        if (isFolder && isExpanded) {
+          handleToggle(event);
+        } else {
+          parentRef?.current.focus();
+        }
+
+        break;
+      }
+      case KeyCode.KEY_RIGHT: {
+        event.preventDefault();
+        if (event.metaKey) { break; }
+
+        if (!isExpanded) {
+          handleToggle(event);
+        } else if (isFolder) {
+          const firstFolderChild = subFolderNode.current.querySelector('[data-item-show-focus=true]');
```

Comment:
If we are querying the DOM, can't we just get the first child li element?

Reply:
We could, but this accomplishes the same thing, and this ensures that we only grab one with the data-item-show-focus, an attribute exclusive to the folder tree component